server/lib/transactions.ts

```diff
@@ -213,6 +215,84 @@ export async function createTransactionsFromPaidExpense(
   return models.Transaction.createDoubleEntry(transaction);
 }
 
+export async function createTransactionsForManuallyPaidExpense(
+  host,
+  expense,
+  paymentProcessorFeeInHostCurrency,
+  totalAmountPaidInHostCurrency,
+  /** Will be stored in transaction.data */
+  transactionData: Record<string, unknown> = {},
+) {
+  assert(paymentProcessorFeeInHostCurrency >= 0, 'Payment processor fee must be positive');
+  assert(totalAmountPaidInHostCurrency >= 0, 'Total amount paid must be positive');
+
+  const isCoveredByPayee = expense.feesPayer === 'PAYEE';
+  const amount = toNegative(totalAmountPaidInHostCurrency - paymentProcessorFeeInHostCurrency);
+  const netAmountInCollectiveCurrency = toNegative(totalAmountPaidInHostCurrency);
+  const amounts = {
+    amount,
+    amountInHostCurrency: amount,
+    paymentProcessorFeeInHostCurrency: toNegative(paymentProcessorFeeInHostCurrency),
+    netAmountInCollectiveCurrency,
+    hostCurrencyFxRate: 1,
+  };
+
+  if (host.currency !== expense.collective.currency) {
+    assert(
+      expense.currency === expense.collective.currency,
+      'Expense currency must be the same as collective currency',
+    );
+    amounts.hostCurrencyFxRate = isCoveredByPayee
+      ? // If the payee is covering the fees, the expense amount should be proportional to the total amount paid in host currency
+        expense.amount / totalAmountPaidInHostCurrency
+      : // If the host is covering the fees, the expense amount be proportional to total amount paid minus the fees
```

Comment:
Note to myself for a second review: aren't these two reversed? I would expect the fees to be ignored in the DEBIT's amount if feesPayer is the PAYEE, and accounted if feesPayer=COLLECTIVE.